### ISSUE TITLE
writeToAutoComplete with SearchView (compat) not works

### ISSUE THREAD

**Issue by alorma** [SchibstedSpain/Barista#244]
Hi @SchibstedSpain/barista-maintainers !

Recently I found the case where we have the following widget on XML:

```
            <android.support.v7.widget.SearchView
                android:id="@+id/searchview_search"
                android:layout_width="match_parent"
                android:layout_height="wrap_content"
                android:gravity="left"
                android:queryHint="@string/search"/>
```

Using `SearchView` is different that normal `EditText` as it extends from `LinearLayoutCompat`

The only way that i found to use this view was the following (using internal ID):

```
writeToAutoComplete(android.support.v7.appcompat.R.id.search_src_text, "Pro")
```

Would be great to have something like:

```
writeToSearchView(R.id.searchview_search, "Pro")
```

What do you think??

**Comment by alorma** [comment]
I can build a PR for this task

**Comment by SmasSive** [comment]
Sounds good for me. I think calling a `writeToAutoComplete` for a `SearchView` is not as fancy as we want for Barista... So IMHO, go for it!

**Comment by Sloy** [comment]
What if we could make `writeToAutoComplete(R.id.searchview_search, "Pro")` automagically work? Would that be better? That way you use the same common API instad of having one ad-hoc.

We could make that work for autocompletes and edittexts by doing some magic with `isDescendantOf()` and `isAssignableFrom()`

**Comment by SmasSive** [comment]
In that case Rafa, I'd change the method name to something more generic. Just to make it more discoverable and intuitive

**Comment by alorma** [comment]
Agree with @SmasSive @Sloy 

`AutoComplete` is not the same as `SearchView`

Maybe we can move it all to `BaristaEditTextInteractions.writeTo()` method?

And the `BaristaEditTextInteractions` will do the magic to check each case (first autocomplete, if no, searchview, if no, normal edit text)

**Comment by Sloy** [comment]
You can use `writeTo` to write into autocompletes also. The name doesn't get much more generic than that 😂 
The difference in `writeToAutoComplete()` is explained in https://github.com/SchibstedSpain/Barista/blob/master/library/src/main/java/com/schibsted/spain/barista/internal/viewaction/AutoCompleteViewActions.java#L33
Basically, it avoids triggering the suggestions, which can get in the way of your tests unless you really want them.

**Comment by alorma** [comment]
I've created the PR for `SearchView` problem.

We can discuss the other problem...

**Comment by alorma** [comment]
Thing is

> We could make that work for autocompletes and edittexts by doing some magic with isDescendantOf() and isAssignableFrom()

SearchView is not descendant of EditText, or AutoComplete

**Comment by Sloy** [comment]
> AutoComplete is not the same as SearchView

I'm not saying it is. What I'm saying is:
If you have a layout structure like
```
<wrapper>
    <editText />
</wrapper>
```

You might want to do `writeTo(wrapper, "text")`. Think of `TextInputLayout` for instance. The same thing happens with the `SearchView`. It's just a wrapper around an EditText. You expect Barista to be smart enough to write into the _internal_ EditText. Right?

That way you avoid knowing the internal id of the SearchView or the TextInputLayout or whatever.

If we agree on that, then it's easy to agree that the current `writeToAutocomplete`  should behave just like `writeTo`,  so it would need to copy the behavior.

Then, @alorma you can use `writeTo` or `writeToAutocomplete` just like now, but using the wrapper id instead of the internal id.

**Comment by Sloy** [comment]
> SearchView is not descendant of EditText, or AutoComplete

That matcher is not about inheritance, but about layout hierarchy instead.
In the example above, `isDescendantOf(wrapper)` would match the EditText.

**Comment by Sloy** [comment]
TL;DR: I think we don't need a specific method for SearchView. Instead, `writeTo()` should magically work.
